matRad_calcLateralParticleCutOff.m

```diff
@@ -42,119 +48,123 @@
-                IntDosePeak     = idd(peakIdx);
-                
+    % get indices for which a lateral cutoff should be calculated
+    cumIntEnergy = cumtrapz(machine.data(energyIx).depths,idd_org);
+    
+    if strcmp(machine.meta.radiationMode,'protons')
+        vEnergySteps        = 0:(cumIntEnergy(end)/NumDepthVal):cumIntEnergy(end);
     else
-        error('unknown dataType');
+        peakTailRelation   = 0.5;
+        NumDepthValToPeak  = ceil(NumDepthVal*peakTailRelation);                                                                          % number of depth values from 0 to peak position
+        NumDepthValTail    = ceil(NumDepthVal*(1-peakTailRelation));                                                                      % number of depth values behind peak position
+        EnergyStepsToPeak  = cumIntEnergy(peakIxOrg)/NumDepthValToPeak;
+        EnergyStepsTail    = (cumIntEnergy(end)-cumIntEnergy(peakIxOrg))/NumDepthValTail;
+        vEnergySteps       = [0:EnergyStepsToPeak:cumIntEnergy(peakIxOrg) cumIntEnergy(peakIxOrg+1):EnergyStepsTail:cumIntEnergy(end)];
     end
     
-    for j = 1:length(Idx)
+    [cumIntEnergy,ix] = unique(cumIntEnergy);
+    depthValues       = matRad_interp1(cumIntEnergy,machine.data(energyIx).depths(ix),vEnergySteps);
+    idd               = matRad_interp1(machine.data(energyIx).depths,idd_org,depthValues);          
+    [~,peakIx]        = max(idd); 
+    
+    cnt = cnt +1 ;
+    % % calculate dose in spot
+    baseData                   = machine.data(energyIx);
+    baseData.LatCutOff.CompFac = 1;   
+  
+    for j = 1:numel(depthValues)
         
         % save depth value
-        machine.data(energyIx).LatCutOff.depths(j) = machine.data(energyIx).depths(Idx(j));
-
-        % calcualate dose for different radi
-        if strcmp(machine.meta.dataType,'singleGauss')
-                    
-                    Sigma1 = sqrt(machine.data(energyIx).sigma(Idx(j))^2 + SigmaIni^2);
-                    dose_r = idd(Idx(j)) * SG(r_mid,Sigma1);
-                  
-        elseif strcmp(machine.meta.dataType,'doubleGauss')
-                    
-                    w      = machine.data(energyIx).weight(Idx(j));
-                    Sigma1 = sqrt(machine.data(energyIx).sigma1(Idx(j)).^2 + SigmaIni^2);
-                    Sigma2 = sqrt(machine.data(energyIx).sigma2(Idx(j)).^2 + SigmaIni^2);
-                    dose_r = DG(r_mid,idd(Idx(j)),w,Sigma1,Sigma2);
-                   
-        else
-            error('unknown dataType');
-        end
-        
+        machine.data(energyIx).LatCutOff.depths(j) = depthValues(j);
+        % calculate dose
+        dose_r         = matRad_calcParticleDoseBixel(depthValues(j) + baseData.offset, radialDist_sq, largestSigmaSq4uniqueEnergies(cnt), baseData);
+       
         if cutOffLevel == 1
             machine.data(energyIx).LatCutOff.CompFac = 1;
             machine.data(energyIx).LatCutOff.CutOff  = Inf;
```

Comment:
shouldn't this read

machine.data(energyIx).LatCutOff.CutOff(j)  = Inf;

and/or go outside of the loop over depths?

Reply:
update: at least for visualization it will result in an error _without_ (j)

Reply:
next update: in calcParticleDose the fine clipping is only performed if there are more than one element... so it basically does not matter. for consistency I would keep it _with_ (j) - if there are no good reasons to do otherwise!

Reply:
I agree - for consistency I would also use the index (j)
machine.data(energyIx).LatCutOff.CompFac(j) = 1;
machine.data(energyIx).LatCutOff.CutOff(j) = Inf;

Reply:
The compensation factor is global, i.e., _no_ (j) there?!

Reply:
yes it is global and not depth dependent. :-)